### ISSUE TITLE
useCreateLineItem not able to refresh the cart 

### ISSUE THREAD

**Issue by TOPSAILL** [medusajs/medusa#9460]
### What Medusa version and documentation are you using?

None

### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

when i use the medusa-react useCreateLineItem function, the cart will not be updated immediately, only after i use the updateCart({email:"xxx@qq.com"}) or some other function in useCart()  can i see the updated line items.   I dont know what is happening here

### How can this issue be resolved?

1.
2.
3.
...

### Are you interested in working on this issue?

- [ ] I would like to fix this issue
